51-android.rules

```diff
@@ -797,91 +797,130 @@ LABEL="not_XiaoMi"
 LABEL="not_Zebra"
 
 # ZTE
 ATTR{idVendor}!="19d2", GOTO="not_ZTE"
 #   ZTE Blade A5 2020
 #   mtp,adb
-ATTR{idProduct}=="0306", ENV{adb_adb}="yes"
+ATTR{idProduct}=="0306", GOTO="adb"
 #   ptp,adb
-ATTR{idProduct}=="0310", ENV{adb_adb}="yes"
+ATTR{idProduct}=="0310", GOTO="adb"
 #   cdrom,adb
-ATTR{idProduct}=="0501", ENV{adb_adb}="yes"
+ATTR{idProduct}=="0501", GOTO="adb"
 #   charging,adb
-ATTR{idProduct}=="1352", ENV{adb_adb}="yes"
+ATTR{idProduct}=="1352", GOTO="adb"
 #   rndis,adb
-ATTR{idProduct}=="1373", ENV{adb_adb}="yes"
+ATTR{idProduct}=="1373", GOTO="adb"
 #   Blade (1353=normal,1351=debug)
-ATTR{idProduct}=="1351", ENV{adb_adb}="yes"
+ATTR{idProduct}=="1351", GOTO="adb"
 #   Blade S (Crescent, Orange San Francisco 2) (1355=normal,1354=debug)
-ATTR{idProduct}=="1354", ENV{adb_adb}="yes"
+ATTR{idProduct}=="1354", GOTO="adb"
 #   P685M LTE modem
-ATTR{idProduct}=="1275", ENV{adb_user}="yes"
+ATTR{idProduct}=="1275", GOTO="user"
 #   MF286[A] internal LTE modem
-ATTR{idProduct}=="1432", ENV{adb_user}="yes"
+ATTR{idProduct}=="1432", GOTO="user"
 #   MF286D internal LTE modem
-ATTR{idProduct}=="1485", ENV{adb_user}="yes"
+ATTR{idProduct}=="1485", GOTO="user"
 #   MF286R internal LTE modem
-ATTR{idProduct}=="1489", ENV{adb_user}="yes"
+ATTR{idProduct}=="1489", GOTO="user"
 #   Nubia / RedMagic Series (NX***)
 #   See https://github.com/TadiT7/nubia_nx619j_dump/blob/NX619J-user-9-PKQ1.180929.001-eng.nubia.20181220.181559-release-keys/vendor/etc/init/hw/init.nubia.usb.rc
 #   ptp,adb
-ATTR{idProduct}=="ffd1", ENV{adb_adb}="yes"
+ATTR{idProduct}=="ffd1", GOTO="adb"
 #   mtp,adb
-ATTR{idProduct}=="ffcf", ENV{adb_adb}="yes"
+ATTR{idProduct}=="ffcf", GOTO="adb"
 #   mass_storage,adb
-ATTR{idProduct}=="ffcd", ENV{adb_adb}="yes"
+ATTR{idProduct}=="ffcd", GOTO="adb"
 #   rndis,adb
-ATTR{idProduct}=="ffcb", ENV{adb_adb}="yes"
+ATTR{idProduct}=="ffcb", GOTO="adb"
 #   modem,service,nema,adb
-ATTR{idProduct}=="ffc9", ENV{adb_adb}="yes"
+ATTR{idProduct}=="ffc9", GOTO="adb"
 #   modem,service,nema,mass_storage,adb
-ATTR{idProduct}=="ffc7", ENV{adb_adb}="yes"
+ATTR{idProduct}=="ffc7", GOTO="adb"
 #   diag,modem,mass_storage,adb
-ATTR{idProduct}=="ffb0", ENV{adb_adb}="yes"
+ATTR{idProduct}=="ffb0", GOTO="adb"
 #   diag,modem,service,mass_storage,adb
-ATTR{idProduct}=="ffb2", ENV{adb_adb}="yes"
+ATTR{idProduct}=="ffb2", GOTO="adb"
 #   adb
-ATTR{idProduct}=="ffc1", ENV{adb_adb}="yes"
+ATTR{idProduct}=="ffc1", GOTO="adb"
 #   diag,mass_storage,adb
-ATTR{idProduct}=="ffc0", ENV{adb_adb}="yes"
-GOTO="android_usb_rule_match"
+ATTR{idProduct}=="ffc0", GOTO="adb"
+GOTO="android_usb_rules_end"
 LABEL="not_ZTE"
 
 # ZUK
-ATTR{idVendor}=="2b4c", ENV{adb_user}="yes"
+ATTR{idVendor}=="2b4c", GOTO="user"
 
 # Verifone
-ATTR{idVendor}=="11ca", ENV{adb_user}="yes"
+ATTR{idVendor}=="11ca", GOTO="user"
+
+# No android devices found
+GOTO="android_usb_rules_end"
+
+# ADB Debug User (default)
+LABEL="adb", ENV{adb_adb}="yes", GOTO="adb_set_values"
+
```

Comment:
I feel like this section would be more readable _without_ those empty lines. Right now it looks as if these were a few more vendors.

Reply:
It looked a bit crunched-up all compressed, but it looks a bit spaced-out with the spaces.
@M0Rf30 - what do you think?

Reply:
> It looked a bit crunched-up all compressed, but it looks a bit spaced-out with the spaces. @M0Rf30 - what do you think?

I like it with blank lines. For now let's focus on making everything work. We leave the aesthetic considerations to a second moment.

Reply:
Sounds good.
I don't like doing a forced push, but have to do a forced push update to keep this PR266 (and avoid creating a new branch in github).

Reply:
you can do every kind of forced push on your branch, it's a very normal stuff (it's a good behavior to do it before opening a PR). I marked this PR as draft, in order to gently finalize it. feel free to mark as ready when you think it's in a good shape both for you and @mskiptr